quodlibet/tests/test_qltk_songlistcolumns.py

```diff
@@ -85,3 +88,20 @@ def test_artist(self):
     def test_people(self):
         column = create_songlist_column("~people")
         self._render_column(column)
+
+    def test_custom_datecol_format(self):
+        format = "%Y%m%d %H:%M:%S PLAINTEXT"
+        quodlibet.config.settext("settings", "datecolumn_timestamp_format",
+                                 format)
+
+        d = datetime.datetime(year=1999, month=5, day=1,
+                              hour=23, minute=11, second=59)
+        stamp = int(time.mktime(d.timetuple()))
+        column = create_songlist_column("~#added")
+        text = self._render_column(column, **{"~#added": stamp})
+        self.assertEqual(text, "19990501 23:11:59 PLAINTEXT")
+
+    def test_nonconfigured_datecol_format(self):
+        text = quodlibet.config.gettext("settings",
+                                        "datecolumn_timestamp_format")
+        self.assertEqual(text, "")
```

Comment:
Perhaps this test (or a separate) should do a quick check to see that the default behaviour _hasn't_ been changed. Bit harder as you said because of the way the existing one works but you could set the date as above and at the very least check that it isn't the above...